Controllers/ToDoController.cs

```diff
@@ -0,0 +1,286 @@
+﻿using RazorPagesAssignment.Data;
+using RazorPagesAssignment.Models;
+using ExcelDataReader;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using X.PagedList;
+
+namespace RazorPagesAssignment.Controllers
+{
+    [Authorize]
+    public class ToDoController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+
+        public ToDoController(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public IActionResult Index(string sortOrder, string searchString, int? page)
+        {
+            ViewData["CurrentSort"] = sortOrder;
+            ViewData["TitleSort"] = String.IsNullOrEmpty(sortOrder) ? "Title" : "title_desc";
+            ViewData["CompletedSort"] = sortOrder == "IsCompleted" ? "IsCompleted" : "completed_desc";
+            ViewData["CreatedSort"] = sortOrder == "CreatedDate" ? "CreatedDate" : "created_desc";
+            ViewData["UpdatedSort"] = sortOrder == "UpdatedDate" ? "UpdatedDate" : "updated_desc";
+
+            var todos = _context.ToDo.ToList();
+
+            if (!string.IsNullOrEmpty(searchString))
+            {
+                todos = todos.Where(t => t.Title.Contains(searchString)).ToList();
+            }
+
+            switch (sortOrder)
+            {
```

Comment:
implement a dictionary or more batter approach instead of multiple switch or if statements

Reply:
Done